lib/core/include/scipp/core/element/bin_detail.h

```diff
@@ -16,9 +16,14 @@ using bin_range_arg =
     std::tuple<scipp::index, scipp::index, Coord, scipp::span<const Edge>>;
 
 static constexpr auto bin_range_common = overloaded{
-    arg_list<bin_range_arg<double, double>, bin_range_arg<int64_t, double>,
-             bin_range_arg<int64_t, int64_t>, bin_range_arg<int32_t, int32_t>,
-             bin_range_arg<int32_t, double>,
+    arg_list<bin_range_arg<double, double>, bin_range_arg<double, float>,
```

Comment:
Added a `TYPED_TEST_SUITE`, it doesn't account for all combinations.

Reply:
I don't think this is testing much: The `bin` algorithm is quite complex, calling a whole series of functions like this one. So in this case this test is mostly just "noise", since it provides no guarantee that the higher-level functionality works.

At a minimum, we should add a (Python) test that tests the full case reported in the linked issue.